test/invalid/idl/typedef-promise.webidl

```diff
@@ -0,0 +1 @@
+typedef (Promise<DOMString> or boolean) Typedef;
```

Comment:
Could you change the filename into `union-promise` because that's the purpose of this patch?